src/utils.js

```diff
@@ -10,9 +10,32 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+import { promises as fs } from 'fs';
 import Async from 'async';
+import Path from 'path';
 
 import { DefaultValues } from './constants';
+import UploadError from './upload-error';
+import ErrorCodes from './error-codes';
+
+const TEMP_PATTERNS = [
+    /^\/~(.*)/, // catch all paths starting with ~
+    /^\/\.(.*)/, // catch all paths starting with .
+];
+
+const TEMP_NAME_PATTERNS = [
+    /^[.~]/i,
+    /^TestFile/, // InDesign: on file open, InDesign creates .dat.nosync* file, renames it TestFile, and deletes it
+    /\.tmp$/i, // Illustrator: on save, creates one or more *.tmp files, renames them to original file name
+    /\.~tmp$/i, // some default Windows applications use this file format
+    // OS X
+    /^DCIM/,
```

Comment:
Maybe also add `.DS_Store`?
On MacOs, DS_Store is a file that stores custom attributes of its containing folder, such as the position of icons or the choice of a background image.

Reply:
Good suggestion. I believe this file would be caught by the first pattern, as it matches anything that begins with a period.